assets/src/edit-story/components/inspector/inspectorProvider.js

```diff
@@ -92,10 +94,13 @@ function InspectorProvider({ children }) {
 
   useEffect(() => {
     tabRef.current = tab;
+  }, [tab]);
+
+  useDeepCompareEffect(() => {
     if (tab === PREPUBLISH) {
       refreshChecklistDebounced();
     }
-  }, [tab, refreshChecklistDebounced, refreshChecklist]);
+  }, [story, refreshChecklistDebounced]);
```

Comment:
So was the issue here that `refreshChecklist` constantly changes, causing this effect to trigger multiple times?

Reply:
That's right

Reply:
You're saying, cache `refreshChecklist` in a ref in `inspectorProvider`?

Reply:
I do want to check the story changing, though, so that when the tab is active the user sees updates to the checklist. We could forgo the package and this part of the code and go for a more manual refresh but that doesn't sound as nice. :)